src/KOKKOS/fft3d_kokkos.cpp

```diff
@@ -147,12 +147,10 @@ struct norm_functor {
 
   KOKKOS_INLINE_FUNCTION
   void operator() (const int &i) const {
-#if defined(FFT_FFTW3) || defined(FFT_CUFFT)
+#if defined(FFT_FFTW3) || defined(FFT_CUFFT) || defined(FFT_MKL)
     FFT_SCALAR* out_ptr = (FFT_SCALAR *)(d_out.data()+i);
     *(out_ptr++) *= norm;
     *(out_ptr++) *= norm;
-#elif defined(FFT_MKL)
```

Comment:
Can we double check this change? I was originally trying to match the non-Hokkos version and https://github.com/lammps/lammps/blob/master/src/KSPACE/fft3d.cpp#L233 I don't think the new version is equivalent.

Reply:
i did a little digging through the source code and found that within LAMMPS the "plan->scaled" flag is always 0 and thus this code does not get used.
@sjplimp do you have some tests for the standalone version of the FFT3d code? One that uses renormalization of the FFTs? Upon superficial examination, the new version in the KOKKOS variant looks correct to me and the MKL non-KOKKOS version seems to be scaling only half the data points.

Reply:
Thanks for looking. Lets just make sure the two versions (Kokkos and non-Kokkos) are consistent.

Reply:
MKL uses the C-style `_Complex` type which may support scaling both real and imaginary in one `*=`, see https://en.cppreference.com/w/c/language/arithmetic_types#Complex_floating_types. Not 100% sure though.

Reply:
"Complex numbers may be used with arithmetic operators + - * and /, possibly mixed with imaginary and real numbers. There are many mathematical functions defined for complex numbers in complex.h"

Reply:
using `d_out(i)` does not compile. It complains about missing arguments. I would have to use the same construct as for KISS. But since the complex data array is just the same as a double array with twice the size, and the complex numbers are multiplied with a real scalar, the effect of both constructs, KOKKOS and KSPACE, should be the same. 
We could just get rid of the ifdefs and only use the KISS version of the loop.